app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -1141,16 +1141,30 @@ protected function _exportCsvItem(Varien_Object $item, Varien_Io_File $adapter)
     /**
      * Retrieve a file container array by grid data as CSV
      *
-     * Return array with keys type and value
+     * Return array with keys type and value [the original way]
+     * Return array with keys file name, type and value, file mime type [the new way]
      *
+     *  the original way (still working): $this->_prepareDownloadResponse($fileName, $grid->getCsvFile($fileName));
+     *      or (with right content type): $this->_prepareDownloadResponse($fileName, $grid->getCsvFile($fileName), 'text/csv');
+     *                       the new way: $this->_prepareDownloadResponse(...$grid->getCsvFile($fileName, -1));
+     *
+     * @param string $fileName
+     * @param int $limit
      * @return array
      * @throws Exception
      */
-    public function getCsvFile()
+    public function getCsvFile($fileName = '', $limit = 0)
     {
         $this->_isExport = true;
         $this->_prepareGrid();
 
+        $count = method_exists($this, 'getCountTotals') ? $this->getCountTotals() : $this->getCollection()->getSize();
```

Comment:
@luigifab phpstan says that method_exists will always return true because the method always exist on this object

Reply:
Mage_Adminhtml_Block_Widget_Grid has the getCountTotals method so it should be right what phpstan says, it's not possible to Mage_Adminhtml_Block_Widget_Grid (or any extended class) to not have the getCountTotals method no?

Reply:
Argh, yes... I don't remember... I can't test with my stupid Android smartphone at the moment, remove it and we will see what happens!